src/services/TeamHub/TeamHub.BLL/Services/MailingService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Net.Mail;
+using System.Text;
+using AutoMapper;
+using Microsoft.Extensions.Options;
+using Shared.SharedModels;
+using TeamHub.BLL.Contracts;
+using TeamHub.BLL.Dtos.TeamMember;
+using TeamHub.DAL.Contracts.Repositories;
+using TeamHub.DAL.Models;
+
+namespace TeamHub.BLL.Services
+{
+    public class MailingService : IMailingService
+    {
+        private readonly IMapper _mapper;
+        private readonly IUserRepository _userRepository;
+        private readonly EmailCredentials _options;
+
+        public MailingService(
+            IMapper mapper,
+            IUserRepository userRepository,
+            IOptions<EmailCredentials> options
+        )
+        {
+            _mapper = mapper;
+            _userRepository = userRepository;
+            _options = options.Value;
+        }
+
+        public async Task SendPendingTasks()
+        {
+            var users = await _userRepository.GetAllAsync();
+
+            using (
+                var client = new CustomSmtpClient(
+                    _options.Host,
+                    _options.Email,
+                    _options.AppPassword
+                )
+            )
+            {
+                foreach (var user in users)
+                {
+                    var teamTeamberResponseDtos = _mapper.Map<List<TeamMemberResponseDto>>(
+                        user.TeamMembers
+                    );
+
+                    var body = ComposeMailBody(user, teamTeamberResponseDtos);
+                    var massage = CreateMailMessage(user.Email, body);
+
+                    try
+                    {
+                        client.Send(massage);
+                    }
+                    catch (Exception ex) { }
```

Comment:
Why are you using a try-catch block here?

Reply:
in case of something went wrong service continues send emails to other clients